### PR TITLE
Unpace path animations and implement paced timing

### PR DESCRIPTION
**Pull request by alancutter**
This pull request includes #310 and changes based on review comments in #310.

### CODE REVIEW

File: web-animations.js [hunk 1]
```diff
@@ -1932,7 +1959,61 @@ StepTimingFunction.prototype = createObject(TimingFunction.prototype, {
...
+    if (segmentLength > 0) {
+      return ((leftIndex + ((length - leftLength) / segmentLength)) /
+          (cumulativeLengths.length - 1));
+    }
+    return leftLength / cumulativeLengths.length;
+  },
+  _cumulativeLengths: function(timedItem) {
+    if (timedItem instanceof Animation &&
+        timedItem.effect instanceof PathAnimationEffect) {
+      this._cumulativeLengths = function () {
```
Comment (dstoc) [inline web-animations.js]: Oh, it is actually stateful.

File: web-animations.js [hunk 2]
```diff
@@ -1932,7 +1964,50 @@ StepTimingFunction.prototype = createObject(TimingFunction.prototype, {
...
+      return ((leftIndex + ((length - leftLength) / segmentLength)) /
+          (cumulativeLengths.length - 1));
+    }
+    return leftLength / cumulativeLengths.length;
+  },
+  _findLeftIndex: function(array, value) {
+    var leftIndex = 0;
+    var rightIndex = array.length - 1;
+    while (rightIndex - leftIndex > 1) {
+      var midIndex = parseInt((leftIndex + rightIndex)/2);
```
Comment (dstoc) [inline web-animations.js]: parseInt -> Math.floor

File: web-animations.js [hunk 3]
```diff
@@ -1932,7 +1964,50 @@ StepTimingFunction.prototype = createObject(TimingFunction.prototype, {
...
+      return 0;
+    }
+    var leftIndex = this._findLeftIndex(cumulativeLengths, length);
+    if (leftIndex >= cumulativeLengths.length - 1) {
+      return 1;
+    }
+    var leftLength = cumulativeLengths[leftIndex];
+    var segmentLength = cumulativeLengths[leftIndex + 1] - leftLength;
+    if (segmentLength > 0) {
+      return ((leftIndex + ((length - leftLength) / segmentLength)) /
```
Comment (dstoc) [inline web-animations.js]: Remove outer most parens

File: web-animations.js [hunk 4]
```diff
@@ -1932,7 +1964,50 @@ StepTimingFunction.prototype = createObject(TimingFunction.prototype, {
...
+  },
+});
+
+/** @constructor */
+var PacedTimingFunction = function(timedItem) {
+  this._timedItem = timedItem;
+};
+
+PacedTimingFunction.prototype = createObject(TimingFunction.prototype, {
+  scaleTime: function(fraction, timedItem) {
```
Comment (dstoc) [inline web-animations.js]: Remove timedItem param

File: web-animations.js [hunk 5]
```diff
@@ -1856,11 +1879,20 @@ var TimingFunction = function() {
...
 
-TimingFunction.createFromString = function(spec) {
+TimingFunction.prototype.scaleTime = abstractMethod;
+
+TimingFunction.createFromString = function(spec, timedItem) {
   var preset = presetTimingFunctions[spec];
   if (preset) {
     return preset;
   }
+  if (spec === "paced") {
```
Comment (dstoc) [inline web-animations.js]: We've been quite inconsistent to date, but we should prefer single quotes for strings (per style guide)

Comment (shans) [inline web-animations.js]: We should lint this when we get a linter.

**Pull request merged**